packages/marble-api/openapis/marblecore-api.yaml

```diff
@@ -3269,6 +3271,126 @@ paths:
           $ref: '#/components/responses/403'
         '404':
           $ref: '#/components/responses/404'
+  /scenario-testrun:
+    get:
+      tags:
+        - TestRun
+      summary: List all test runs for a scenario
+      operationId: listTestRuns
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: scenarioId
+          description: ID of the scenario that need to be fetched
+          in: query
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: List of corresponding test runs
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+    post:
+      tags:
+        - TestRun
+      summary: Create a test run
+      operationId: createTestRun
+      security:
+        - bearerAuth: []
+      requestBody:
+        description: 'Test run object that needs to be added to the database'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TestRunCreateInputDto'
+        required: true
+      responses:
+        '200':
+          description: Created test run
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '422':
+          $ref: '#/components/responses/422'
+  /scenario-testruns/{testRunId}:
+    get:
+      tags:
+        - TestRun
+      summary: Get a test run by id
+      operationId: getTestRun
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: testRunId
+          description: ID of the test run that need to be fetched
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: The test run corresponding to the provided `testRunId`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
```

Comment:
Required for the testrun detail page.
The current endpoint is not supposed to return graph data (the APi for this is out of scope of this PR, existing work in [this openned backend PR](https://github.com/checkmarble/marble-backend/pull/746))

Reply:
done .. I will integrate it